modules/segmented-control/react/stories/stories.tsx

```diff
@@ -16,32 +16,36 @@ import {SegmentedControl} from '../index';
 
 import README from '../README.md';
 
-storiesOf('Labs|Segmented Control/React', module)
-  .addParameters({component: SegmentedControl})
-  .addDecorator(withReadme(README))
-  .add('Default', () => {
-    const [value, setValue] = React.useState<string | number>();
-    const handleToggle = (selectedValue: string | number) => {
-      setValue(selectedValue);
-      action('Segmented Control selection change')(selectedValue);
-    };
+export default {
+  title: 'Components|Buttons/Segmented Control/React',
+  component: SegmentedControl,
+  decorators: [withReadme(README)],
+};
 
-    return (
-      <SegmentedControl value={value} onChange={handleToggle}>
-        <IconButton
-          icon={listViewIcon}
-          value="list-view"
-          aria-label="List View"
-          onClick={e => action('Existing IconButton onClick callback')(e)}
-        />
-        <IconButton
-          icon={worksheetsIcon}
-          value="table-view"
-          aria-label="Table View"
-          disabled={true}
-        />
-        <IconButton icon={deviceTabletIcon} value="device-view" aria-label="Device View" />
-        <IconButton icon={percentageIcon} value="percent-view" aria-label="Percent View" />
-      </SegmentedControl>
-    );
-  });
+export const Default = () => {
+  const [value, setValue] = React.useState<string | number>();
+  const handleToggle = (selectedValue: string | number) => {
+    setValue(selectedValue);
+    action('Segmented Control selection change')(selectedValue);
+  };
+
+  return (
+    <SegmentedControl value={value} onChange={handleToggle}>
+      <IconButton
+        icon={listViewIcon}
+        value="list-view"
+        aria-label="List View"
+        // Note: For some reason this action call causes a delay in storybook
```

Comment:
I'll run some analysis on this

Reply:
Initial investigation shows terrible performance around strigifying a large JSON object. I'll have to pull down locally to get a proper stack trace though

Reply:
OMG. What is Storybook doing?! Clicking is trigging an action which is running through `JSON.stringify` a bunch of times with rather huge objects and then posing a message across the iframe.

![Storybook_slowness](https://user-images.githubusercontent.com/338257/84426497-92138c00-abe0-11ea-866d-05d1d6ed49f6.gif)

Reply:
<img width="992" alt="Screen Shot 2020-06-11 at 12 51 33 PM" src="https://user-images.githubusercontent.com/338257/84428580-f2f09380-abe3-11ea-984d-b0667e29e8bb.png">

Storybook seems wildly inefficient

Reply:
Removing the 2 `action` calls drastically reduces the time spent in Storybook code, but there's still a lot that Storybook is doing, decreasing any performance of our stories. This makes it very difficult to measure meaningful performance of our components.

Reply:
Well that's a bummer. We use `action` in quite a few places. Are you proposing we remove it here? It's bad, but it's not terribly bad 🤷‍♂️

Reply:
I think we'll save that for when we redo documentation.